scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,434 @@
+	token *proto.AttestationToken,
+	trustAnchors []string,
+	endorsementsStrings []string,
+) error {
+	var (
+		taEndorsement handler.Endorsement
+		avk           comid.KeyTriple
+		tsm           tokens.TSMReport
+	)
+
+	// Get the ARK TA
+	for i, t := range trustAnchors {
+		var endorsement handler.Endorsement
+
+		if err := json.Unmarshal([]byte(t), &endorsement); err != nil {
+			return fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
+		}
+
+		if endorsement.Type == handler.EndorsementType_VERIFICATION_KEY {
+			taEndorsement = endorsement
+			break
+		}
+	}
+
+	if taEndorsement.Type != handler.EndorsementType_VERIFICATION_KEY {
+		return fmt.Errorf("trust anchors unavailable")
+	}
+
+	err := json.Unmarshal(taEndorsement.Attributes, &avk)
+	if err != nil {
+		return err
+	}
+
+	provisionedArk := avk.VerifKeys[0]
+
+	// Parse certificate chain in evidence (auxblob)
+	err = tsm.FromCBOR(token.Data)
+	if err != nil {
+		return err
+	}
+
+	protoReport, err := abi.ReportToProto(tsm.OutBlob)
+	if err != nil {
+		return err
+	}
+
+	ark, err := getARK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	arkBlock, err := readCert(ark)
+	if err != nil {
+		return err
+	}
+
+	ask, err := getASK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	askBlock, err := readCert(ask)
+	if err != nil {
+		return err
+	}
+
+	vcek, err := getVCEK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	vcekBlock, err := readCert(vcek)
+	if err != nil {
+		return err
+	}
+
+	// Test if TA matches with the one supplied in evidence
+	match, err := compareTAs([]byte(provisionedArk.String()), ark)
+	if err != nil {
+		return err
+	}
+	if !match {
+		return fmt.Errorf("ARK in evidence does not match provisioned ARK")
+	}
+
+	// Validate the integrity of evidence by ensuring the
+	// certificate chain is intact, and the signature is valid
+	var attestation sevsnp.Attestation
+	attestation.Report = protoReport
+	attestation.CertificateChain = &sevsnp.CertificateChain{VcekCert: vcekBlock, AskCert: askBlock, ArkCert: arkBlock}
+	err = verify.SnpAttestation(&attestation, snpAttestationOptions())
+
+	if err != nil {
+		log.Errorf("failed to validate certificate chain: %+v\n", err)
+	}
+
+	return err
+}
+
+// refvalToComidTriple converts extracted reference values to CoMID value triple
+func refvalToComidTriple(endorsementsStrings []string) (*comid.ValueTriple, error) {
+	var (
+		refValEndorsement handler.Endorsement
+		rv                comid.ValueTriple
+	)
+
+	for i, e := range endorsementsStrings {
+		var endorsement handler.Endorsement
+
+		if err := json.Unmarshal([]byte(e), &endorsement); err != nil {
+			return nil, fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
+		}
+
+		if endorsement.Type == handler.EndorsementType_REFERENCE_VALUE {
+			refValEndorsement = endorsement
+			break
+		}
+	}
+
+	if refValEndorsement.Type != handler.EndorsementType_REFERENCE_VALUE {
+		return nil, fmt.Errorf("reference values unavailable")
```

Comment:
Errorf without formatting is wasteful. Better to have a global var like `ErrRefValUnavailable = errors.New("reference values unavailable")` so a caller can compare for equality if needed.

Reply:
OK, I'll add a global variable so a caller can compare for equality.